plugins/by-name/flutter-tools/settings-options.nix

```diff
@@ -3,81 +3,102 @@ let
   inherit (lib) types;
   inherit (lib.nixvim)
     defaultNullOpts
-    literalLua
     mkNullOrOption'
     mkNullOrOption
     ;
 in
 {
   ui = {
-    border = defaultNullOpts.mkNullable (with types; either str (listOf (maybeRaw str))) "rounded" ''
-      The border type to use for all floating windows, the same options/formats used for
-      `:h nvim_open_win` e.g. `"single"` | `"shadow"` | `[<list-of-eight-chars>]`.
-    '';
+    border = defaultNullOpts.mkBorder' {
+      pluginDefault = "single";
+      example = "rounded";
+      name = "all floating windows";
+    };
   };
+
   decorations = {
     statusline = {
       app_version = defaultNullOpts.mkBool false ''
         Set to true to be able use the `flutter_tools_decorations.app_version` in your statusline.
+
         This will show the current version of the flutter app from the `pubspec.yaml` file.
       '';
 
       device = defaultNullOpts.mkBool false ''
         Set to true to be able use the `flutter_tools_decorations.device` in your statusline.
+
         This will show the currently running device if an application was started with a specific
         device.
       '';
 
       project_config = defaultNullOpts.mkBool false ''
         Set to true to be able use the `flutter_tools_decorations.project_config` in your
-        statusline this will show the currently selected project configuration.
+        statusline.
+
+        This will show the currently selected project configuration.
       '';
     };
   };
 
   debugger = {
     enabled = defaultNullOpts.mkBool false ''
-      Integrate with nvim `dap` + install dart code debugger.
+      Enable `nvim-dap` integration.
     '';
 
-    exception_breakpoints = defaultNullOpts.mkListOf types.anything [ ] ''
-      If empty `dap` will not stop on any exceptions, otherwise it will stop on those specified.
-      See `|:help dap.set_exception_breakpoints()|` for more info.
+    exception_breakpoints = defaultNullOpts.mkListOf types.anything null ''
+      If empty dap will not stop on any exceptions, otherwise it will stop on those specified.
+
+      See `|:help dap.set_exception_breakpoints()|` for more information.
     '';
 
     evaluate_to_string_in_debug_views = defaultNullOpts.mkBool true ''
       Whether to call `toString()` on objects in debug views like hovers and the variables list.
+
       Invoking `toString()` has a performance cost and may introduce side-effects, although users
-      may expected this functionality.
+      may expect this functionality.
+
+      `null` is treated like `false`.
```

Comment:
This could get confusing, due to our null-culling.

In practice, `null` is treated as undefined, so the plugin default (`true`) is used.

Maybe this could be rephrased to refer to `mkRaw "nil"`? Or maybe the plugin default is incorrect?